docs/gcp_setup.md

```diff
@@ -0,0 +1,287 @@
+--permissions=bigquery.tables.getData,\
+bigquery.tables.updateData \
+--stage=GA
+
+echo "GRANT \`projects/${PROJECT}/roles/goe_bq_app_role\` ON SCHEMA ${TARGET_DATASET}
+TO \"serviceAccount:${SVC_ACCOUNT}@${PROJECT}.iam.gserviceaccount.com\";
+" | bq query --project_id=${PROJECT} --nouse_legacy_sql --location=${LOCATION}
+```
+
+If GOE is permitted to create tables then add these extra privileges:
+```
+gcloud iam roles update goe_bq_app_role --project ${PROJECT} \
+--add-permissions=bigquery.tables.create,bigquery.tables.update
+```
+
+If GOE is permitted to drop tables then add these extra privileges:
+```
+gcloud iam roles update goe_bq_app_role --project ${PROJECT} \
+--add-permissions=bigquery.tables.delete
+```
+
+#### goe_bq_stg_role
+Note that the role grant is bound to the staging BigQuery dataset (which has the same name as the target dataset but with an "_load" suffix), no project wide access is granted. The `bq` utility is used to grant the role because `gcloud` does not support these granular grants.
+
+Also note that the staging dataset must be created *before* executing these commands (see [Generating Dataset DDL](#generating-dataset-ddl) for details).
+```
+gcloud iam roles create goe_bq_stg_role --project ${PROJECT} \
+--title="GOE Data Staging Access" \
+--description="Grants GOE permissions to manage objects in a staging schema" \
+--permissions=bigquery.tables.create,\
+bigquery.tables.delete,\
+bigquery.tables.getData \
+--stage=GA
+
+echo "GRANT \`projects/${PROJECT}/roles/goe_bq_stg_role\` ON SCHEMA ${TARGET_DATASET}_load
+TO \"serviceAccount:${SVC_ACCOUNT}@${PROJECT}.iam.gserviceaccount.com\";
+" | bq query --project_id=${PROJECT} --nouse_legacy_sql --location=${LOCATION}
+```
+
+#### goe_dataproc_role
+```
+gcloud iam roles create goe_dataproc_role --project ${PROJECT} \
+--title="GOE Dataproc Access" --description="GOE Dataproc Access" \
+--permissions=dataproc.clusters.get,dataproc.clusters.use,\
+dataproc.jobs.create,dataproc.jobs.get,\
+iam.serviceAccounts.getAccessToken \
+--stage=GA
+
+gcloud projects add-iam-policy-binding ${PROJECT} \
+--member=serviceAccount:${SVC_ACCOUNT}@${PROJECT}.iam.gserviceaccount.com \
+--role=projects/${PROJECT}/roles/goe_dataproc_role
+
+gcloud projects add-iam-policy-binding ${PROJECT} \
+--member=serviceAccount:${SVC_ACCOUNT}@${PROJECT}.iam.gserviceaccount.com \
+--role=roles/iam.serviceAccountUser
+```
+
+#### goe_batches_role
+```
+gcloud iam roles create goe_batches_role --project ${PROJECT} \
+--title="GOE Dataproc Access" --description="GOE Dataproc Access" \
+--permissions=dataproc.batches.create,dataproc.batches.get \
+--stage=GA
+
+gcloud projects add-iam-policy-binding ${PROJECT} \
+--member=serviceAccount:${SVC_ACCOUNT}@${PROJECT}.iam.gserviceaccount.com \
+--role=projects/${PROJECT}/roles/goe_batches_role
+```
+
+## Compute Engine Virtual Machine
+Values supplied below are examples only, changes will likely be required for each use case:
```

Comment:
...examples only. Changes are likely to be...

Reply:
Fixed